internals/daemon/access.go

```diff
@@ -68,3 +68,18 @@ func (ac UserAccess) CheckAccess(d *Daemon, r *http.Request, user *UserState) Re
 	// An identity explicitly set to "access: untrusted" isn't allowed.
 	return Unauthorized(accessDenied)
 }
+
+// MetricsAccess allows requests over the HTTP from authenticated users.
+type MetricsAccess struct{}
+
+func (ac MetricsAccess) CheckAccess(d *Daemon, r *http.Request, user *UserState) Response {
+	if user == nil {
+		return Unauthorized(accessDenied)
+	}
+	switch user.Access {
+	case state.MetricsAccess, state.ReadAccess, state.AdminAccess:
+		return nil
+	}
```

Comment:
Nit: I think it could be nice to put the code below in a `default` condition, that way we can force the compiler to complain if a new codepath doesn't return a value. Although the above `switch` is short so it isn't too big of an issue here, being explicit could be a nice precedent